fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/patient/ChunkReader.java

```diff
@@ -110,18 +121,23 @@ public ChunkReader() {
 
     private void fillChunkDataBuffer(List<Resource> resources) throws Exception {
         TransientUserData chunkData = (TransientUserData) jobContext.getTransientUserData();
+        int indexOfCurrentTypeFilter = 0;
         int compartmentPageNum = 1;
         int resSubTotal = 0;
         FHIRSearchContext searchContext;
-        Class<? extends Resource> resourceType = ModelSupport
-                .getResourceType(resourceTypes.get(indexOfCurrentResourceType));
+        Class<? extends Resource> resourceType = ModelSupport.getResourceType(resourceTypes.get(indexOfCurrentResourceType));
+
+        if (searchParametersForResoureTypes == null ) {
+            searchParametersForResoureTypes = BulkDataUtils.getSearchParemetersFromTypeFilters(fhirTypeFilters);
+        }
+
         if (chunkData != null) {
-            for (Resource res : resources) {
-                if (res == null) {
-                    continue;
-                }
-                Patient patient = (Patient) res;
+            do {
                 Map<String, List<String>> queryParameters = new HashMap<>();
+                // Add the search parameters from the current typeFilter for current resource type.
+                if (searchParametersForResoureTypes.get(resourceType) != null) {
+                    queryParameters.putAll(searchParametersForResoureTypes.get(resourceType).get(indexOfCurrentTypeFilter));
+                }
                 List<String> searchCreteria = new ArrayList<>();
```

Comment:
```suggestion
                List<String> searchCriteria = new ArrayList<>();
```

Reply:
ha, good catch, let me refactor in my eclipse.

Reply:
done